source/includes/dev/_microservice_dev_guide.md

```diff
@@ -195,6 +200,88 @@ Running tests under Docker is similar to running the actual microservice. The ea
+
+Depending on the scale of the changes you're doing, it's recommended to use [git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) with [semantic versioning](https://semver.org/). Also be sure to update the `CHANGELOG.md` accordingly, and the `package.json` or equivalent files if they refer a version number. It's also best practice to announce the changes you're about to deploy before doing so, so that other developers of RW API applications can be on the lookout for regressions, and can quickly get in touch with you should any undesired behavior change be detected.
+
+Each of the referred environments lives on a separate [Kubernetes](https://kubernetes.io/) cluster (hosted with [AWS EKS](https://aws.amazon.com/eks/)), and deployment is done using individual Jenkins instances:
+
+- [Jenkins for the dev environment](https://jenkins.aws-dev.resourcewatch.org)
+- [Jenkins for the staging environment](https://jenkins.aws-staging.resourcewatch.org)
+- [Jenkins for the production environment](https://jenkins.aws-prod.resourcewatch.org)
+
+All 3 instances have similar overall configuration, but different microservices may deploy differently depending on the behavior coded into the `Jenkinsfile` that's part of their source code - for example, some WRI sites are also deployed using this approach, but opt to deploy both staging and production versions to the `production` cluster, and may not be in the `staging` or `dev` Jenkins.
+
+The list of jobs you find on each Jenkins instance will match the list of services deployed on that environment. In the details of each job, you should find a branch named after the environment, which corresponds to the Github branch with the same name (some services may still have the old approach, with `develop` for `dev` and `staging`, and `master` for `production`). You may also find other branches, or a different branch structure, depending on the service itself - again, the `Jenkinsfile` configuration is king here, and you should refer to it to better understand what is the desired behavior per branch. In some cases, old branches will be listed on Jenkins but should be ignored.
+
+Once you start a deployment process, Jenkins will run the `Jenkinsfile` content - it is, after all, a script - and perform the actions contained in it. While it's up to the maintainer of each microservice to modify this script, more often that not it will run the tests included in the microservice, using Docker, and if these pass, push the newly generated Docker image to [Docker Hub](https://hub.docker.com/). It will then update the respective Kubernetes cluster with content of the matching subfolder inside the `k8s` folder of the microservice, plus the `k8s/service` folder if one exists. The last step is to deploy the recently pushed Docker image from Docker Hub to the cluster, which will cause Kubernetes to progressively replace running old instances of the service with ones based on the new version.
```

Comment:
Not clear whether I actually have to trigger a build here or if that happens automatically once the PR is merged.